apps/desktop/src/app/upload/select-files/select-files.component.ts

```diff
@@ -0,0 +1,102 @@
+import { 
+  Component,
+  AfterViewInit,
+  OnDestroy,
+  Input,
+  Output,
+  EventEmitter,
+  ViewChild,
+  ElementRef,
+  ChangeDetectorRef,
+  ChangeDetectionStrategy
+} from '@angular/core';
+import { SelectionModel } from '@angular/cdk/collections';
+import { concat, merge, fromEvent, Observable, Subject, of } from 'rxjs';
+import { tap, filter, takeUntil, auditTime, scan, switchMap, concatMap, share } from 'rxjs/operators';
+import { SyncItem, ISourceConnector, SearchResults } from '../../sync/models';
+
+@Component({
+  selector: 'da-select-files',
+  templateUrl: './select-files.component.html',
+  styleUrls: ['./select-files.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SelectFilesComponent implements AfterViewInit, OnDestroy {
+  @Input() source: ISourceConnector | undefined;
+  @Input() selection: SelectionModel<SyncItem>;
+  @Output() selectionChange = new EventEmitter<SelectionModel<SyncItem>>();
+  @Output() next = new EventEmitter<void>();
+  @ViewChild('scroll') private scroll?: ElementRef;
+
+  query = '';
+  triggerSearch = new Subject<void>();
+  triggerNextPage = new Subject<void>();
+  unsubscribeAll = new Subject<void>();
+  nextPage?: Observable<SearchResults>;
+  loading = false; 
+
+  items: SyncItem[] = [];
+  resources: Observable<SyncItem[]> = this.triggerSearch.pipe(
+    filter(() => !!this.source),
+    tap(() => { this.loading = true }),
+    switchMap(() => 
+      concat(
+        (this.source as ISourceConnector).getFiles(this.query),
+        this.triggerNextPage.pipe(
+          filter(() => !this.loading),
+          tap(() => { this.loading = true }),
+          concatMap(() => this.nextPage ? this.nextPage : of({ items: [], nextPage: undefined }))
+        )
+      ).pipe(
+        tap((res) => { this.nextPage =  res.nextPage; this.loading = false; }),
+        scan((acc, current) => acc.concat(current.items), [] as SyncItem[])
+      ),
+    ),
+    share()
+  );
+
+  constructor(private cdr: ChangeDetectorRef) { }
+
+  ngAfterViewInit() {
+    setTimeout(() => { this.triggerSearch.next() }, 200);
+
+    // TODO: can't use async pipe in the template because (unexpectedly) it doesn't work on GDrive connector.
```

Comment:
I had some weird things with GDrive too, not sure exactly what it is about, but it is not critical anyway.